php_phongo.c

```diff
@@ -507,7 +518,10 @@ zend_bool phongo_writeerror_init(zval *return_value, bson_t *bson TSRMLS_DC) /*
 		writeerror->code = bson_iter_int32(&iter);
 	}
 	if (bson_iter_init_find(&iter, bson, "errmsg") && BSON_ITER_HOLDS_UTF8(&iter)) {
-		writeerror->message = bson_iter_dup_utf8(&iter, NULL);
+		uint32_t errmsg_len;
+		const char *err_msg = bson_iter_utf8(&iter, &errmsg_len);
+
+		writeerror->message = estrndup(err_msg, errmsg_len);
```

Comment:
I don't understand the bug... nor the fix.


http://mongoc.org/libbson/current/bson_iter_dup_utf8.html

Yes, this string is supposed to be free()d with bson_free (), but that is the same as efree() because we set bson allocators to the same functions.

Wouldn't it be easier to keep this as is, and use bson_free () correctly, rather then bulky code to copy it?